engine/src/engine.rs

```diff
@@ -1920,27 +1961,23 @@ impl<'env, I: IO + Copy, E: Env, M: ModExpAlgorithm> Backend for Engine<'env, I,
         result
     }
 
-    /// Check is storage empty for the address
-    /// EIP-7610: non-empty storage
+    /// Check if the storage of the address is empty.
+    /// Related to EIP-7610: non-empty storage
     fn is_empty_storage(&self, address: H160) -> bool {
         let address = Address::new(address);
-        // As we can't read all storage data for account we assuming that if storage exists
-        // `index = 0` always true
-        let index = H256::zero();
-        let generation = *self
-            .generation_cache
-            .borrow_mut()
-            .entry(address)
-            .or_insert_with(|| get_generation(&self.io, &address));
-        // First we're checking cache to not produce read-storage operation
         if self
             .contract_storage_cache
             .borrow()
-            .contains_key(&(address, index))
+            .contains_key(&(address, H256::zero()))
         {
-            return false;
+            return true;
```

Comment:
```suggestion
            return false;
```

How does this incorrect logic pass our CI? Maybe new test that touches `is_empty_storage` is needed.